code/modules/mob/living/simple_animal/hostile/bees.dm

```diff
@@ -179,10 +179,22 @@
 	else
 		. = ..()
 		if(. && beegent && isliving(target))
-			var/mob/living/L = target
-			if(L.reagents)
-				beegent.expose_mob(L, INJECT)
-				L.reagents.add_reagent(beegent.type, rand(1,5))
+			on_sting(target)
+
+/**
+ * Handles exposing the target mob to the bees venom and injecting the target with the venom.
+ *
+ * Arguments:
+ * - [target][/mob/living]: The mob that has been stung.
+ */
+/mob/living/simple_animal/hostile/bee/proc/on_sting(mob/living/target)
+	var/sting_amt = rand(1, 5)
+
+	target.reagents?.add_reagent(beegent.type, sting_amt)
+	target.expose_reagents(list((beegent) = sting_amt), null, INJECT, volume_modifier = 1, show_message = TRUE)
+	if (target.reagents)
+		beegent.on_transfer(target, INJECT, sting_amt)
```

Comment:
Calling this proc is managed by the `trans_to` proc normally and using it externally for a particular feature is a bad sign. If you need this to be called you should be moving the reagents using the usual procs.